modules/trainer/GenericTrainer.py

```diff
@@ -518,17 +531,31 @@ def sample_commands_fun():
                     self.model.optimizer.zero_grad(set_to_none=True)
                     has_gradient = False
 
+                    last_lr = lr_scheduler.get_last_lr()[0]
+                    scheduled_lr = last_lr
+                    if self.__adaptive_optimizer:
+                        last_lr *= self.model.optimizer.param_groups[0]["d"]
+                    unet_lr = self.config.unet.learning_rate if self.config.unet.learning_rate else self.config.learning_rate
```

Comment:
Here you assume that the model always has a unet and a single text encoder. This is only true for SD1.2 and SD2.1 models. The implementation needs to be more generic.

Reply:
Done. Moved it to the model setup class, which is unique per type of training and thus dispatch on it will work for any module.